src/main/java/tars/Parser.java

```diff
@@ -0,0 +1,95 @@
+package tars;
+
+import java.util.ArrayList;
+/**
+ * Processes User inputs and executes commands based on inputs
+ * Has method to also list all Tasks added by User
+ *
+ * @author csk
+ * @version 1
+ */
+
+public class Parser {
+    private static final String LINE = "    _____________________________________________";
+
+    public Parser() {
+    }
+    /**
+     * Checks user inputs after being taken as string and string[] variables
+     * Chooses appropriate methods to call based on input given or throws error message
+     * depending on incorrect input received
+     *
+     * @param entryParts
+     * @param entry
+     * @param tasks
+     */
+    public String checkEntry(String[] entryParts, String entry, TaskList tasks) {
+        String result = "";
+        try {
+            if (entry.equals("todo") || entry.equals("deadLINE") || entry.equals("event")) {
+                throw new TarsException(LINE + "\n" + "    OOPS! Describe the task/event/deadLINE/todo or list"
+                        + "\n" + LINE);
+            } else if (entryParts[0].equals("list")) {
+                result = this.listPrint(tasks);
+            } else {
+                throw new TarsException(LINE + "\n" + "    OOPS! Only accept a task/event/deadLINE/todo as input"
+                        + "\n" + LINE);
+            }
+        } catch (TarsException e) {
+            result = e.getMessage();
+        }
+        return result;
+    }
+
+    /**
+     * Prints all the Tasks in the list at the current state of storage
+     *
+     * @param tasks
+     */
+    public String listPrint(TaskList tasks) {
```

Comment:
I think this method's name is supposed to be ``printList`` instead, right? i.e. in the verb form 😺